pkg/util/block_source_reader.go

```diff
@@ -0,0 +1,138 @@
+package util
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io"
+	"sync"
+)
+
+/*
+ A BlockSource is a data source that reads/writes in block size increments.
+ */
+type BlockSource interface {
+	Read(startSector uint64, numSectors uint64, buffer []byte) (sectorsRead uint64, err error)
```

Comment:
It looks more a ReadAt func based on its param list.

Reply:
OK, I updated it